packages/syft/src/syft/store/kv_document_store.py

```diff
@@ -396,11 +398,15 @@ def _update(
                 )
 
                 # update the object with new data
-                for key, value in obj.to_dict(exclude_empty=True).items():
-                    if key == "id":
-                        # protected field
-                        continue
-                    setattr(_original_obj, key, value)
+                if overwrite:
```

Comment:
This basically overtires the key with the new object instead of just updating the values that are not set.